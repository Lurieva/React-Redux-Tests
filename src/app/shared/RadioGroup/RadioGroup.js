import React from 'react';
import PropTypes from 'prop-types';

const RadioGroup = ({ name, param, options, title, onChange, groupClassName }) => (  
    <div className={groupClassName}>
        <span className={`${groupClassName}__title`}>{title}</span>
        <div className="btn-group" onChange={onChange}>
            {options.map(option => {
                const { value, label } = option;
                const isCurrent = param === value;
 
                return (
                    <label key={value} 
                           className={`btn ${isCurrent ? 'radio-active' : '' }`}>
                        <input type="radio" 
                               name={name}
                               value={value}
                               defaultChecked={isCurrent}/> {label}
                    </label>
                );
            })}
        </div>
    </div>
);

RadioGroup.propTypes = {  
    title: PropTypes.string,
    name: PropTypes.string.isRequired,
    options: PropTypes.array.isRequired,
    onChange: PropTypes.func.isRequired,
    param: PropTypes.string.isRequired,
    groupClassName: PropTypes.string
};

export default RadioGroup;