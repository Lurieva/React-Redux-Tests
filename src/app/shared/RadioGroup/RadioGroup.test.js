import React from 'react';
import { shallow } from 'enzyme';
import RadioGroup from './RadioGroup.js';

describe('RadioGroup', () => {
    it('should be defined', () => {
        expect(RadioGroup).toBeDefined();
    });
});