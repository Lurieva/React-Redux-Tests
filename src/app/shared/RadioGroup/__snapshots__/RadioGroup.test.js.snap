// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup should render correctly  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioGroup
    groupClassName="classNameMock"
    name="radio"
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "option": "opt",
          "value": "value",
        },
      ]
    }
    param="value"
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="classNameMock__title"
        >
          title
        </span>,
        <div
          className="btn-group"
          onChange={[MockFunction]}
        >
          <label
            className="btn radio-active"
          >
            <input
              defaultChecked={true}
              name="radio"
              type="radio"
              value="value"
            />
             
          </label>
        </div>,
      ],
      "className": "classNameMock",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "title",
          "className": "classNameMock__title",
        },
        "ref": null,
        "rendered": "title",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="btn radio-active"
            >
              <input
                defaultChecked={true}
                name="radio"
                type="radio"
                value="value"
              />
               
            </label>,
          ],
          "className": "btn-group",
          "onChange": [MockFunction],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "value",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  defaultChecked={true}
                  name="radio"
                  type="radio"
                  value="value"
                />,
                " ",
                undefined,
              ],
              "className": "btn radio-active",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "defaultChecked": true,
                  "name": "radio",
                  "type": "radio",
                  "value": "value",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              " ",
              undefined,
            ],
            "type": "label",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="classNameMock__title"
          >
            title
          </span>,
          <div
            className="btn-group"
            onChange={[MockFunction]}
          >
            <label
              className="btn radio-active"
            >
              <input
                defaultChecked={true}
                name="radio"
                type="radio"
                value="value"
              />
               
            </label>
          </div>,
        ],
        "className": "classNameMock",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "title",
            "className": "classNameMock__title",
          },
          "ref": null,
          "rendered": "title",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="btn radio-active"
              >
                <input
                  defaultChecked={true}
                  name="radio"
                  type="radio"
                  value="value"
                />
                 
              </label>,
            ],
            "className": "btn-group",
            "onChange": [MockFunction],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "value",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    defaultChecked={true}
                    name="radio"
                    type="radio"
                    value="value"
                  />,
                  " ",
                  undefined,
                ],
                "className": "btn radio-active",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "defaultChecked": true,
                    "name": "radio",
                    "type": "radio",
                    "value": "value",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                " ",
                undefined,
              ],
              "type": "label",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
